src/fitnesse/services/PageService.java

```diff
@@ -0,0 +1,31 @@
+package fitnesse.services;
+
+import fitnesse.FitNesseContext;
+import fitnesse.http.Request;
+import fitnesse.http.Response;
+import fitnesse.responders.NotFoundResponder;
+import fitnesse.wiki.PathParser;
+import fitnesse.wiki.WikiPage;
+import fitnesse.wiki.WikiPagePath;
+
+public class PageService {
```

Comment:
Instead of creating this new service, maybe it would be better to see whether you can make the responders you modified extend `BasicResponder`. It provides a `makeResponse()` implementation that seems to map nicely on what is needed for these responders including a `pageNotFoundResponse()` method....